### PR TITLE
[Backport 1.3] Add plugin install workflow and action 

### PR DESCRIPTION
**Pull request by scrawfor99**
Signed-off-by: Stephen Crawford <steecraw@amazon.com>

### Description
Backport the plugin install workflow and github action to 1.3. The backport got missed when everything else was merged into one branch. 

Backports merged PR #2271 to 1.3 branch. 

### Testing
Tested for functionality on Java and Windows with JDK 8, 11, 14. It is worth noting that the plugin install workflow seems slightly flaky on jdk 11 alone. I encountered this issue first when I was originally merging PR #2271. For whatever reason the setup.bat file does not always echo the  'yes' response into the install script as expected. It passes over 50% of the time though. 

### Check List
- [x] New functionality includes testing
- [x] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/plugin_install.yml
Comment (RyanL1997) [inline .github/workflows/plugin_install.yml:22]: Just for my knowledge, is this line the representation of the setup of JDK? I know we are using this type of matrix to support multi platforms/versions.

Comment (scrawfor99) [reply .github/workflows/plugin_install.yml:22]: Yeah it is exactly the same as the platform stuff. Basically just works like a for loop.

File: .github/actions/start-opensearch-with-one-plugin/action.yml
Comment (peternied) [reply .github/actions/start-opensearch-with-one-plugin/action.yml:33]: I'd rather we don't attempt to review/update style comments in pull requests like number of blank lines.  If you'd like to see a type of style enforced - I'd look into a tool that can enforce/correct the style towards a preference.

Checkout actions in the github marketplace [1] that could prettify/lint yml.

[1] https://github.com/marketplace/actions/yaml-lint

**Review by peternied — APPROVED** [review]

**Review by cwperks — APPROVED** [review]

File: .github/actions/start-opensearch-with-one-plugin/action.yml
Comment (scrawfor99) [reply .github/actions/start-opensearch-with-one-plugin/action.yml:33]: Should I remove it or leave it? I can do either.

**Comment by cwperks** [comment]
Changed the branch protections to now make the `plugin-install` checks required instead of `linux-install` and `windows-install`. I added the requirement only for JDK11, but that can be changed if needed.

**Comment by peternied** [comment]
As soon as the option to merge is green you can merge, no matter any outstanding comments.

Conversely, if there are any changes maintainers would like before a PR is merged they should use 'Request Changes' to ensure that gets resolved before the code goes in.

**Comment by cwperks** [comment]
@scrawfor99 Should this change be backported to the 2.x line as well? I only see it in main and 1.3 now.

**Pull request merged**
